docs/inference_protocols.md

```diff
@@ -97,17 +99,200 @@ These options can be used to configure the KeepAlive settings:
+the major version that you compiled against, and the minor version
+provided by the shared library is greater-than-or-equal to the minor
+version that you compiled against, then your application can use the
+Server API.
+
+```
+#include "tritonserver.h"
+// Error checking removed for clarity...
+uint32_t api_version_major, api_version_minor;
+TRITONSERVER_ApiVersion(&api_version_major, &api_version_minor);
+if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
+    (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
+  // Error, the shared library implementing the Server API is older than
+  // the version of the Server API that you compiled against.
+}
+```
+
+#### Non-Inference APIs
+
+The Server API contains functions for checking health and readiness,
+getting model information, getting model statistics and metrics,
+loading and unloading models, etc. The use of these functions is
+straightforward and some of these functions are demonstrated in
+[simple.cc](../src/simple.cc) and all are documented in
+[tritonserver.h](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h).
+
+#### Inference APIs
+
+Performing an inference request requires the use of many Server API
+functions and objects, as demonstrated in
+[simple.cc](../src/simple.cc). The general usage requires the
+following steps.
+
+* Create a `TRITONSERVER_ResponseAllocator` using
+  `TRITONSERVER_ResponseAllocatorNew`.  You can use the same response
+  allocator for all of your inference requests, or you can create
+  multiple response allocators.  When Triton produces an output
+  tensor, it needs a memory buffer into which it can store the
+  contents of that tensor. Triton defers the allocation of these
+  output buffers by invoking callback functions in your
+  application. You communicate these callback functions to Triton with
+  the `TRITONSERVER_ResponseAllocator` object. You must implement two
+  callback functions, one for buffer allocation and one for buffer
+  free. The signatures for these functions are
+  `TRITONSERVER_ResponseAllocatorAllocFn_t` and
+  `TRITONSERVER_ResponseAllocatorReleaseFn_t` as defined in
+  [tritonserver.h](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h). In
+  [simple.cc](../src/simple.cc), these callback functions are
+  implemented as `ResponseAlloc` and `ResponseRelease`.
+
+* Create an inference request as a `TRITONSERVER_InferenceRequest`
+  object. The inference request is where you specify what model you
+  want to use, the input tensors and their values, the output tensors
+  that you want returned, and other request parameters. You create an
+  inference request using `TRITONSERVER_InferenceRequestNew`. You
+  create each input tensor in the request using
+  `TRITONSERVER_InferenceRequestAddInput` and set the data for the
+  input tensor using
+  `TRITONSERVER_InferenceRequestAppendInputData`. By default, Triton
```

Comment:
Should other `AppendInputDataXXX` APIs be mentioned? Something like check tritonserver.h for other variances